src/gather_classes.rs

```diff
@@ -257,80 +255,131 @@ fn update_report(
+                    let native_list = diff_map
+                        .get("native")
+                        .ok_or(anyhow!("difference should have native"))?;
+
+                    let mut base_calls = Vec::new();
+                    get_calls_inner(base_list, &mut base_calls)?;
+                    let mut native_calls = Vec::new();
+                    get_calls_inner(native_list, &mut native_calls)?;
+
+                    result.extend(merge_calls_with_count(base_calls, native_calls));
+                } else {
+                    match get_call_with_count(obj) {
+                        Ok(current_call) => {
+                            result.push(current_call);
+                        }
+                        // We let the parsing fail silently, as there may be inner calls that are valid
+                        Err(err) => {
+                            debug!("Failed to extract tuple with error: {err}");
+                        }
+                    }
+
+                    if let Some(calls_value) = obj_map.get("calls") {
+                        get_calls_inner(calls_value, result)?;
+                    }
+                }
+                Ok(())
             }
+            _ => Err(anyhow!("unexpected value: `{}`", obj)),
         }
-        _ => Err(anyhow!("unexpected value: `{}`", obj)),
     }
+
+    let mut result = Vec::new();
+    get_calls_inner(obj, &mut result)?;
+    Ok(result)
 }
 
-/// returns `Result<(EntryPoint, ClassHash), anyhow::Error>`
-fn get_tuple_from_call(call: &Value) -> Result<(EntryPoint, ClassHash), anyhow::Error> {
-    let mut values = Vec::<FieldElement>::new();
-    for key in ["entry_point_selector", "class_hash"] {
-        let next = call
-            .get(key)
-            .context(format!("call: `{call}`\n-------Extracting key {key}"))?;
-        let felt_version = serde_json::from_value(next.clone())
-            .context(format!("convert value `{}` to felt", next))?;
-        values.push(felt_version);
+/// Retrieves a [CallWithCount] from a call object.
+///
+/// If any of the keys are Different, then the call is considered invalid.
+fn get_call_with_count(call: &Value) -> Result<CallWithCount, anyhow::Error> {
+    let keys = ["entry_point_selector", "class_hash"];
+    let mut values = Vec::with_capacity(keys.len());
+
+    for key in keys {
+        if let Some(Value::String(string)) = call.get(key) {
+            let value = if string_is_same(string) {
+                parse_same_string(string).context("Failed to parse SAME value")?
+            } else {
+                string
+            };
+
+            let felt_version = FieldElement::from_hex_be(value)
+                .context(format!("Failed to convert value to felt"))?;
+            values.push(felt_version);
+        } else {
+            return Err(anyhow!("Failed to parse call {call} for key: {key}"));
+        }
     }
-    Ok((values[0], values[1]))
+
+    Ok(((values[0], values[1]), 1))
```

Comment:
This is a bit hard to read. Let's improve readability of the code so that by just reading one can understand what's being returned there.

Reply:
I made the method more functional, I think it's clearer to understand at first glance now